tests/common/test_preprocessors.py

```diff
@@ -0,0 +1,198 @@
+"""Unit tests for TLU optimization preprocessors."""
```

Comment:
a bit like above, looks great overall but very few (random) comments that makes the file hard to understand (several times I have no clue what is actually going on)